### PR TITLE
Story25 segmenting

### PR DESCRIPTION
**Pull request by Pontius1007**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Description

Added functionality to split  road segments on meter and/or minimum data points for each segment. The whole road network is then added to the database. 

Added pagination for API and new endpoints needed for roadsegments. 

Added tests for the code. 

## Motivation and Context

Solves Task #80 

## How Has This Been Tested?

Look at included tests. 

## Checklist:

- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Known buggs: 
For some reason some elements are created with 0 datapoints. These are filtered out and not added to the database.

### CODE REVIEW

File: backend/data/road_segmenting/calculate_distance.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    """
+
+    length = 0
+    index = 0
+    for gps_point in gps_list:
+        index += 1
+        if gps_list.index(gps_point) > 0:
+            prev = gps_list[gps_list.index(gps_point) - 1]
+            if harvesine:
+                # todo: make sure the harvesine formula returns in meters
```
Comment (sardred) [inline backend/data/road_segmenting/calculate_distance.py]: Was this done?

File: backend/api/permissions.py [hunk 2]
```diff
@@ -12,7 +12,7 @@ def has_permission(self, request, view):
         else:
             return request.user.is_staff
 
-
+
```
Comment (sardred) [inline backend/api/permissions.py]: 👾

File: backend/data/road_segmenting/calculate_distance.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
+from math import *
```
Comment (sardred) [inline backend/data/road_segmenting/calculate_distance.py]: I'm sure Travis is already complaining about this, but [don't do this](https://stackoverflow.com/questions/2386714/why-is-import-bad?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa).
Alternatives would be `import math`, so you can call `ceil(x)` with `math.ceil(x)`
You can also shorten it (even though math is plenty short) by writing `import math as m`, which would let you run `m.ceil(x)`, though in this case I'd just use math without an alias.

File: backend/data/road_segmenting/calculate_distance.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    Calculates the distance between two points using the distance formula
+    :param start_list: the first GPS point, given in UTM list form [start_x, start_y]
+    :param end_list:  the second GPS point, given in UTM list form [start_x, start_y]
+    :return: The distance between the first GPS point and the second GPS point
+    """
+    distance = sqrt(pow((start_list[0] - end_list[0]), 2) + pow((start_list[1] - end_list[1]), 2))
+    return distance
+
+
+def harvesine_formula(start_list, end_list):
```
Comment (sardred) [inline backend/data/road_segmenting/calculate_distance.py]: Do a project-wide rename here, the name is [Haversine](https://en.wikipedia.org/wiki/Haversine_formula), not Harvesine.

File: backend/data/road_segmenting/example_roadnet_to_db.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    filtered_road['to_meter']           = road['properties']['til_meter']
+    filtered_road['typeofroad']         = road['properties']['typeVeg']
+    filtered_road['roadsection']        = road['properties']['vegavdeling']
+    filtered_road['roadsectionid']      = road['properties']['veglenkeid']
+    filtered_road['vrefshortform']      = road['properties']['vrefkortform']
+
+    return filtered_road
+
+
+def format_to_db(kommune, type_road, max_distance, min_segments):
```
Comment (sardred) [inline backend/data/road_segmenting/example_roadnet_to_db.py]: Does 'kommune' here need to be norwegian? Or can a lot of these variables be anglicized?

File: backend/data/road_segmenting/test_segmenting.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        '''
+        margin = 50
+        error = 0
+        for road in self.split_segments:
+            if (road['properties']['strekningslengde'] - self.max_segment_length) > margin:
+                error += 1
+        print(str(error) + ' veier gikk over grensen i lengde')
+        self.assertLess(error, 20)
+
+    # test post to db
```
Comment (sardred) [inline backend/data/road_segmenting/test_segmenting.py]: Will this be filled in?

File: backend/data/road_segmenting/test_segmenting.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+    # test deletion from db
```
Comment (sardred) [inline backend/data/road_segmenting/test_segmenting.py]: See above

File: backend/data/road_segmenting/calculate_distance.py (hunk 1 above)
Comment (Pontius1007) [reply backend/data/road_segmenting/calculate_distance.py]: Yes

Comment (sardred) [reply backend/data/road_segmenting/calculate_distance.py]: Then please remove the comment in the cleanup push :)

**Pull request merged**